gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -147,6 +158,19 @@ private InstantiatingGrpcChannelProvider(Builder builder) {
             : builder.directPathServiceConfig;
   }
 
+  /**
+   * Package-Private constructor that is only visible for testing DirectPath functionality inside
+   * tests. This overrides the computed systemProductName when the class is initialized to help
+   * configure the result of {@link #isOnComputeEngine()} check.
+   *
+   * <p>If productName is null, that represents the result of an IOException
+   */
+  @VisibleForTesting
+  InstantiatingGrpcChannelProvider(Builder builder, String productName) {
```

Comment:
Adding the EnvProvider as another param would need to make the envProvider variable non-final. I kept the package-private setter for now until we can migrate to using junit-pioneer to set the env vars.